extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/error/MSQWarningReportLimiterPublisher.java

```diff
@@ -74,12 +78,22 @@ public void publishException(int stageNumber, Throwable e)
       totalCount = totalCount + 1;
       errorCodeToCurrentCount.compute(errorCode, (ignored, count) -> count == null ? 1L : count + 1);
 
+      // Send the warning as an error if it is disallowed altogether
+      if (disallowedWarningCode.contains(errorCode)) {
+        try {
+          controllerClient.postWorkerError(workerId, MSQErrorReport.fromException(workerId, host, stageNumber, e));
+        }
+        catch (IOException e2) {
```

Comment:
Can we throw a new RE with the message `failed to post the worker error xyz to the controller`. That way the worker will get terminated with a relevant error.

Reply:
Nit: rename from e2 to postException?